### DISCUSSION TITLE
`Dispatchers.Main.immediate` for navigation. A safe choice?

### DISCUSSION THREAD

**Discussion by YektaDev** [arkivanov/Decompose#496] [Q&A] [answered]
Hi👋.

I wonder if it's safe to always use **`Dispatchers.Main.immediate`** in case of navigation (particularly _stack_ navigation, if it makes any difference). I'm asking to know if I can wrap a `withContext(Dispatchers.Main.immediate) {...}` over my abstraction functions that are related to navigation, so I don't need to worry about being on the main thread in domain; be it inside a coroutine or a synchronous execution flow.

**Comment by arkivanov** [comment, marked as answer]
Thanks for the question! I think it should be fine and safe.
